frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/builders/vm/NumaUnitToVmBaseBuilder.java

```diff
@@ -24,8 +25,13 @@ protected void build(UnitVmModel model, VmBase vm) {
             return;
         }
 
+        boolean isNumaPinningAllowed = model.getNumaSupportCommand().getIsExecutionAllowed();
+
+        boolean isResizeAndPinDropped = model.getOriginalCpuPinningPolicy() == CpuPinningPolicy.RESIZE_AND_PIN_NUMA
+                && model.getCpuPinningPolicy().getSelectedItem().getPolicy() != CpuPinningPolicy.RESIZE_AND_PIN_NUMA;
+
         List<VmNumaNode> nodeList = model.getVmNumaNodes();
-        if (nodeList == null || nodeList.size() != nodeCount) {
+        if (nodeList == null || nodeList.size() != nodeCount || !isNumaPinningAllowed || isResizeAndPinDropped) {
```

Comment:
does it mean that if we had resize-and-pin and then drop it, only the pinning would be dropped but the nodes would remain? that's a bit unexpected because the user doesn't set the amount of nodes when using resize-and-pin, right?

not that important though..

Reply:
This is needed to clear the Numa CPU pinnings that would otherwise caused the validation to fail.

Reply:
> This is needed to clear the Numa CPU pinnings that would otherwise caused the validation to fail.

do I get this right that you mean the nodes should have been dropped as well but it's technically a complicated change to for the UI to cope with and so we do it as a hack to avoid that?